src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.Validation.cs

```diff
@@ -78,7 +79,7 @@ static Utf16Utility()
             long tempUtf8CodeUnitCountAdjustment = 0;
             int tempScalarCountAdjustment = 0;
 
-            if (Sse2.IsSupported)
+            if (AdvSimd.Arm64.IsSupported || Sse2.IsSupported)
```

Comment:
As a safety mechanism, may want to introduce a `Debug.Assert(BitConverter.IsLittleEndian)` check within this block. It wasn't needed with x86 / SSE2, but ARM could in theory run on a big-endian platform. And IIRC the bit twiddling in this code assumes little-endian.

Reply:
Reasonable ask, considering I had the same check in my other PRs.